### ISSUE TITLE
Plugin compatible with version 4.26

### ISSUE THREAD

**Issue by lcjefferson** [leapmotion/LeapUnreal#71]
I'm trying to open in version 4.26 of unreal, however, there is an error. Did they have an updated version of the plugin?

**Comment by SebastienTussiau** [comment]
Up

**Comment by z3099644** [comment]
I'm also very interested in a plugin compatible with 4.26

**Comment by 1157457050** [comment]
Up

**Comment by omidghajar** [comment]
Hey sorry guys for my late reply and posting how to rebuild and compile the Plugin, I was busy and caught up with some projects,

Straight to the point, I will be explaining step by step with pictures for ease of understanding, **How to Rebuild Leap Motion Plugin in Unreal Engine 4.25 or 4.26** successfully.

1- First you need to go to the link and download **Microsoft Visual Studio 2019** [,](https://visualstudio.microsoft.com/downloads/) then make sure you download the community version as it is free, upon installation, you will prompt to tick what kind of installation components you want to install, and follow the picture which is idle for Unreal Engine.
![Step (1)](https://user-images.githubusercontent.com/73571955/106023770-13814580-6102-11eb-98fe-f5c43a81a4db.jpg)

2- Second, after installation, download the latest Leap Motion Plugin, in your project folder, create a folder calls "**Plugins**" and then extract the LeapUnreal-3.6.0.zip file into the "**Plugins**" folder.
![Step (2)](https://user-images.githubusercontent.com/73571955/106023998-4aeff200-6102-11eb-8e05-f421aa07eae6.jpg)

3- After extractions, click on the project file to open, then you will prompt with the rebuilding the plugin, click on **Yes** to begin the rebuilding process.
![Step (3)](https://user-images.githubusercontent.com/73571955/106024151-71ae2880-6102-11eb-9c50-ce6b40c9ee3b.jpg)

4- You need to wait for the rebuild process, you do need to do anything, just sit back, you can click on **Show Log** to see the rebuilding process if you like to, it might take a while depends on your system.
![Step (4)](https://user-images.githubusercontent.com/73571955/106024322-a0c49a00-6102-11eb-8e44-9c18ffbbead9.jpg)

5- Once the rebuild process is done, the project automatically will be opened as the picture below
![Step (5)](https://user-images.githubusercontent.com/73571955/106024437-c0f45900-6102-11eb-9bdc-52948b6561e0.jpg)

6- **BINGO!** Unreal Engine has rebuilt the plugin successfully, as you tick show plugin content, you can find Leap Motion Content folder in under you plugin content.
![Step (6)](https://user-images.githubusercontent.com/73571955/106024776-13357a00-6103-11eb-9da6-dc22a418d22f.jpg)

7- I suggest to close Unreal and connect your leap motion and then reopen your project and you are ready to configure and use your beautiful leap motion controller, you can check the "**How to use it - Convenience Rigged Pawn**" following the link at [](https://github.com/leapmotion/LeapUnreal)

PS:  If you have a VR setup alongside your Leap Motion Controller, you can start with **Option 1.** VR Mode- Example Pawn and test your Leap Motion Controller and VR set in VR Mode, if you do not have VR set, you can still use **Option 2.** Desktop Mode - Example Actor to begin to use your Leap Motion Controller without using VR mode in Normal Play Mode or Simulation Mode.

You can check out the test video of Realtime Leap Motion Controller in Unreal Engine.

https://user-images.githubusercontent.com/73571955/106026153-6a881a00-6104-11eb-93c2-cb69bd655da8.mp4



Hope this could help you to kickstart your amazing journey with Leap Motion Controller and Unreal Engine.

Cheers 👍 

Regards.
Omid.

**Comment by SebastienTussiau** [comment]
Thank you! Will try that!

**Comment by lcjefferson** [comment]
Thanks!

**Comment by MaxPalmer-UH** [comment]
Thanks @omidghajar for your input and helping others. We haven't got an official 4.26 release done yet, but the current master contents should be compatible with 4.26 and contains a few other fixes. As it's not part of an official release (i.e. not recommended for production) it hasn't been through an official QA test process but might be a good starting point for people.

**Comment by grifu** [comment]
Thanks for this guide.
I've followed all steps with Visual Studio Community 2019, but still, LeapMotion returns an error while compiling in Unreal 4.26.1, with this error: LeapMotion could not be compied. try rebuilding from source manually. 
After starting the project it  prompts for rebuilding the plugin, then it starts building
It returns the following error:
LeapImage (83): error C2027: use of undefined type 'FTexture2DResource'

**Comment by getnamo** [comment]
Replace FTexture2DResource with FTextureResource for 4.26 compatibility fix

**Comment by ando9527** [comment]
> Replace FTexture2DResource with FTextureResource for 4.26 compatibility fix

I got another issue after replacement.

'GetCurrentFirstMip': is not a member of 'FTextureResource'

**Comment by getnamo** [comment]
See https://github.com/getnamo/LeapUnreal/commit/73c0f504b1ed3880a54346e0d0531f982138088e#diff-66eb947d81d9f03f4a0080690e22fa50bde8265fe05d7b00f095dec44d26557d for ref fixes

**Comment by JimSimons-UH** [comment]
@lcjefferson this is now fixed so closing the issue.

**Comment by SebastienTussiau** [comment]
I still get a "fatal error" message when building the .exe of the game in the Shinpping configuration in UE4.26 Does anyone have this problem?

**Comment by omidghajar** [comment]
> I still get a "fatal error" message when building the .exe of the game in the Shinpping configuration in UE4.26 Does anyone have this problem?

As @MaxPalmer-UH said and I quote "not recommended for production) it hasn't been through an official QA test process" so it is advisable not to use it yet for shipping the product, but maybe he would be able to guide you for this issue.

**Comment by SebastienTussiau** [comment]
> > I still get a "fatal error" message when building the .exe of the game in the Shinpping configuration in UE4.26 Does anyone have this problem?
> 
> As @MaxPalmer-UH said and I quote "not recommended for production) it hasn't been through an official QA test process" so it is advisable not to use it yet for shipping the product, but maybe he would be able to guide you for this issue.

Thank you for your reply @omidghajar , @MaxPalmer-UH  when should we expect an official release of the plugin for ue4.26?

**Comment by JimSimons-UH** [comment]
Tracking this now in the other issue https://github.com/leapmotion/LeapUnreal/issues/77.

**Comment by wilbercl** [comment]
Hello, I need the UnrealCV plugin for Unreal Engine 4.27, I'm trying to connect to the server of a videogame engine (binary game) (.exe) and it gives me the following error:

```
ERROR:__init__:205:Can not connect to ('localhost', 9000)
ERROR:__init__:206:Error [WinError 10061] No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión
ERROR:__init__:153:Fail to send message, client is not connected
```
Thanks
